### ISSUE TITLE
[Feature request] Windows Hello support

### ISSUE THREAD

**Issue by Avik-B** [keeweb/keeweb#961]
Is it possible to add windows hello support for log in?

**Comment by antelle** [comment]
Hi! What is "windows hello"?

**Comment by antelle** [comment]
Sorry, stupid question indeed, found it in google.
It's possible, as well as Apple Fingerprint, but probably would be quite complicated.

**Comment by BynariStar** [comment]
Should be quite simple once [WebAuthn](https://www.w3.org/TR/2018/CR-webauthn-20180320/) is fully implemented into browsers. If I'm not mistaken, browsers currently support hardware security keys, but not biometric authentication just yet.

**Comment by Bodup** [comment]
It's not a matter of authentication however. We want Windows Hello to store our master-key in the TPM, WebAuthn can't do that. 
That said I'm dying to find a KeePass client for Windows that implements this. There's really no need to type complicated passphrases all the time if it can simply be stored in hardware. 
If one could unlock their database using biometrics/a pin then the passphrase would only need to be typed once you use a new device, allowing for much more complicated passphrases to be used as well.

**Comment by vdurante** [comment]
Could somebody point me to a straightforward documentation on how to use Windows Hello to store  the password? I am willing to try implementing it, but last time I tried I couldn't make it work using C# and MSDN docs

**Comment by antelle** [comment]
@vdurante I don’t think there’s support in Electron now

**Comment by vdurante** [comment]
@antelle I know it doesn't, but I was thinking about either calling external executable or using something like [edge](https://github.com/tjanczuk/edge)

**Comment by antelle** [comment]
I don't think that calling an external executable to compute a piece of master key is a good idea.

**Comment by vdurante** [comment]
Why?

Sent from Outlook Mobile<https://aka.ms/blhgte>

________________________________

**Comment by antelle** [comment]
It's easier to attack, for example, replace the executable with another one. We can of course check it before running, but still, having a library would be better.

**Comment by vdurante** [comment]
Either way, I believe we can call a few C# functions for managing windows hello from node. Will attempt doing something when I am free.

Sent from Outlook Mobile<https://aka.ms/blhgte>

________________________________

**Comment by kbarnawi** [comment]
> Could somebody point me to a straightforward documentation on how to use Windows Hello to store the password? I am willing to try implementing it, but last time I tried I couldn't make it work using C# and MSDN docs

This App in Microsoft store have windows Hello implementation on it 
https://www.microsoft.com/en-au/p/keepassreader/9nblggh4s820?activetab=pivot:overviewtab

**Comment by kbarnawi** [comment]
> Sorry, stupid question indeed, found it in google.
> It's possible, as well as Apple Fingerprint, but probably would be quite complicated.

I loved this app but biometric log in Is critical for me to begin using it so sad still does not support Windows hello 😔

**Comment by kbarnawi** [comment]
> Should be quite simple once [WebAuthn](https://www.w3.org/TR/2018/CR-webauthn-20180320/) is fully implemented into browsers. If I'm not mistaken, browsers currently support hardware security keys, but not biometric authentication just yet.

I think chrome and Edge both support biometric

**Comment by antelle** [comment]
WebAuthn won't help here, it's a bit different thing.  
There's also a blocker described in #679.

**Comment by antelle** [comment]
As soon as this is resolved, we'll be able to support this, as well as macOS touch id, since now we can have native modules and the integration should be pretty easy, for example using [this](https://github.com/MarkusJx/node-ms-passport#readme) or similar packages. But unfortunately now it's a no-go, since you don't want that all apps running on your computer even without special privileges have access to your passwords.

**Comment by kbarnawi** [comment]
> This App in Microsoft store have windows Hello implementation on it
> https://www.microsoft.com/en-au/p/keepassreader/9nblggh4s820?activetab=pivot:overviewtab

@antelle so is this app is implementation is not sure?  I am not developer :)

**Comment by antelle** [comment]
It's hard to explain to a non-developer if you don't have a lot of time for that, but I'll try do describe it: we 're using [Electron](https://www.electronjs.org/) as a framework for KeeWeb. Electron has a very serious limitation which makes KeeWeb vulnerable if we implement things like TouchID or Windows Hello.

It's not a vulnerability in Electron and it's not a vulnerability in KeeWeb, however together with TouchID / Windows Hello it's a significant security risk.

**Comment by antelle** [comment]
I posted it in the linked issue about Touch ID, I made a (rather bad) workaround for the Electron issues mentioned above, for reference, it's [here](https://github.com/antelle/electron-evil-feature-patcher). So once Electron v12 is released, we'll be able to implement this as well.

**Comment by vdurante** [comment]
@antelle thank you so much for taking your time to check into things and give us some feedback. We are happy to know you are tracking this issue and willing to implement it. I hope Electron supports it in the future so we can get past this barrier. Happy hollidays!

**Comment by colomboe** [comment]
Hello, is there any update on this topic? I see that touch ID support has been released some months ago and keeweb is now on electron 12 too. Is this something coming soon?

**Comment by antelle** [comment]
I don't think it's supported in Electron yet, but I don't know for sure.

**Comment by matiaskee** [comment]
Hi, is there any update on this? would be a great feature :)

**Comment by Swepool** [comment]
This would be nice to have :)

**Comment by tejonaco** [comment]
Any update on this?

**Comment by Ladbaby** [comment]
For anyone who may want a Windows Hello supported KeeWeb, my fork implement it: https://github.com/Ladbaby/keeweb/releases

DISCLAIMER: Feature is implemented via vibe coding. I have tried to fix some possible security problems, but since I'm not familiar with electron best practices, it comes with no guarantee. Use at your own risk. If you discovery any possible vulnerability, feel free to tell me.
